### ISSUE TITLE
How does the calibration method find the best threshold for the eye frame?

### ISSUE THREAD

**Issue by crissallan** [antoinelame/GazeTracking#35]
Nice job, however I'd like to know how does the calibration method works for finding the best threshold? Because when I use a near infrared camera to detect the iris, I find that the contour of iris became the contour of the whole eye region, so it leads to a wrong location of Iris when I calculated the mass center of iris.
Is there any suggestion for this?
Many thanks!

**Comment by antoinelame** [comment]
Hello @crissallan,

To avoid users to provide a threshold value themselves, I created an automatic calibration algorithm to find the right threshold value for the user/webcam.

I did some statistics and noticed that the size of the iris is always about 48% of the surface of the eye for all people (when they are looking to the center). Thresholds values to binarize images can differ a lot from person to person, but iris sizes are very stable.

However, I may not have done enough statistics on different types of eyes. Maybe for Asian eyes, the 48% size is slightly inaccurate. But I'm really not sure, I'd have to check it. I don't think the difference is significant, though.

So on the current version, here's how works the automatic calibration algorithm:

- During the first 20 frames, I send them to the Calibration class
- I try to binarize the frame with different thresholds values from 5 to 100 (with a step of 5) and I calculate iris sizes
- For each frame, I save the value that gives the closest iris size to 48%
- When the first 20 frames are analyzed, the final threshold value is the average of the best 20 values

[More information there](https://github.com/antoinelame/GazeTracking/issues/5#issuecomment-472078559)!
